packages/extension-api-explorer/src/ExtensionApiExplorer.tsx

```diff
@@ -84,17 +69,52 @@ const configurator = new ExtensionConfigurator()
 export const ExtensionApiExplorer: FC = () => {
   const match = useRouteMatch<{ specKey: string }>(`/:specKey`)
   const extensionContext = useContext<ExtensionContextData>(ExtensionContext)
+  const [specs, setSpecs] = useState<SpecItems>()
 
-  let chosenSdk: any
+  let sdk: any
   if (match?.params.specKey === '3.1') {
-    chosenSdk = extensionContext.core31SDK
-  } else if (match?.params.specKey === '4.0') {
-    chosenSdk = extensionContext.core40SDK
+    sdk = extensionContext.core31SDK
+  } else {
+    sdk = extensionContext.core40SDK
+  }
+
+  /** fetch and compile an API specification to an ApiModel
+   *
+   * @param spec to fetch and compile
+   */
+  async function extFetch(spec: SpecItem) {
+    if (!spec.specURL) return undefined
+    const sdk = extensionContext.core40SDK
+    const [version, name] = spec.specURL?.split('/').slice(-2)
+    const content = await sdk.ok(sdk.api_spec(version, name))
+    // TODO figure out why this crazy step is required
+    let json = JSON.parse(content)
+    if (typeof json === 'string') {
+      json = JSON.parse(json)
+    }
+    json = upgradeSpecObject(json)
+    const api = ApiModel.fromJson(json)
+    return api
   }
 
+  useEffect(() => {
+    /** Load Looker /versions information and retrieve all supported specs */
+    async function getSpecs() {
+      const versions = await sdk.ok(sdk.versions())
+      return getSpecsFromVersions(versions, (spec) => extFetch(spec))
+    }
+
+    getSpecs()
+      .then((resp) => setSpecs(resp))
+      .catch((err) => console.error(err))
+  }, [sdk])
+
   return (
-    <RunItProvider sdk={chosenSdk} configurator={configurator} basePath="">
-      <ApiExplorer specs={specs} />
+    <RunItProvider sdk={sdk} configurator={configurator} basePath="">
+      <>
+        {specs && <ApiExplorer specs={specs} />}
+        {!specs && 'Loading API specifications from Looker ...'}
```

Comment:
This can probably be made more attractive but it's a theme provider/component provider issue and it very briefly appears anyway

Reply:
AFAICT, you can't use any Looker components at that level without embedding ComponentProvider. I suppose I could do that.

Reply:
Doh, assumed it was higher up in the component tree. I would consider moving it. Downside is that it impacts the standalone version as well. I dont think that is too bad though